js/tokens.js

```diff
@@ -0,0 +1,496 @@
+        return parsedOrderbook;
+    }
+
+    async fetchTicker (symbol, params = { 'time': 'hour' }) {
+        await this.loadMarkets ();
+        let ticker = await this.publicGetPublicTickerPair (this.extend ({
+            'pair': this.marketId (symbol),
+        }, params));
+        let timestamp = parseInt (ticker['timestamp']) * 1000;
+        let vwap = this.safeFloat (ticker, 'vwap');
+        let baseVolume = this.safeFloat (ticker, 'volume');
+        let quoteVolume = undefined;
+        if (baseVolume !== undefined && vwap !== undefined)
+            quoteVolume = baseVolume * vwap;
+        let last = this.safeFloat (ticker, 'last');
+        return {
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeFloat (ticker, 'high'),
+            'low': this.safeFloat (ticker, 'low'),
+            'bid': this.safeFloat (ticker, 'bid'),
+            'bidVolume': undefined,
+            'ask': this.safeFloat (ticker, 'ask'),
+            'askVolume': undefined,
+            'vwap': vwap,
+            'open': this.safeFloat (ticker, 'open'),
+            'close': last,
+            'last': last,
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': baseVolume,
+            'quoteVolume': quoteVolume,
+            'info': ticker,
+        };
+    }
+
+    nonce () {
+        return this.milliseconds ();
+    }
+
+    parseTrades (trades, market = undefined, since = undefined, limit = undefined) {
+        if (trades.trades.length === 0) {
+            return [];
+        }
+        let result = Object.values (trades.trades).map (trade => this.parseTrade (trade, market));
+        result = this.sortBy (result, 'timestamp');
+        let symbol = (market !== undefined) ? market['symbol'] : undefined;
+        return this.filterBySymbolSinceLimit (result, symbol, since, limit);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = { 'time': 'hour' }) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let response = await this.publicGetPublicTradesTimePair (this.extend ({
+            'pair': market['id'],
+        }, params));
+        return this.parseTrades (response, market, since, limit);
+    }
+
+    async fetchBalance (currency = undefined) {
+        await this.loadMarkets ();
+        let result = {};
+        if (typeof currency === 'undefined') {
+            result['info'] = [];
+            const keys = Object.keys (this.currencies);
+            for (let i = 0; i < keys.length; i += 1) {
+            // for (let key in this.currencies) {
+                let res = await this.privateGetPrivateBalanceCurrency ({ 'currency': keys[i] });
+                let account = this.account ();
+                account['free'] = parseFloat (res.available);
```

Comment:
Unfortunately, this won't transpile, because it violates one of the rules – all associative keys should be in quotes: `res.bad`, `res['good']`. Can you edit for this throughout the file?

Here's the list of rules that we need to follow precisely: 

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#multilanguage-support
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes